### PR TITLE
Add a release guide

### PR DESCRIPTION
**Pull request by ghelmling**

### CODE REVIEW

File: RELEASING.rst [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+performing a release.
+
+
+Performing the Release
+======================
+
+Ensure Local Branch is Up-to-date
+---------------------------------
+
+First, make sure your local copy of the ``develop`` branch is up to date with all changes::
```
Comment (awholegunch) [inline RELEASING.rst]: up-to-date

File: RELEASING.rst [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+Release the Staging Repository in Artifactory
+---------------------------------------------
+
+Release the artifact bundle in Artifactory:
+
+1. Login to https://oss.sonatype.org (you will need to use the same credentials you have
+   configured in your ``~/.m2/settings.xml`` file) 
+2. Go to "Staging Repos"
+3. Find the "comcontinuuity" repo with the Tephra release.  Be sure to expand the contents of the
```
Comment (awholegunch) [inline RELEASING.rst]: should that be "com.continuuity" ?

File: RELEASING.rst [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+   repo to confirm that it contains the correct Tephra artifacts. 
+4. Click on "Release" button at top, and enter a brief description, such as "Tephra N.N.N release"
```
Comment (awholegunch) [inline RELEASING.rst]: Click on the "Release" button

File: RELEASING.rst [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+This will checkout the source code using the release tag, build the release and deploy it to the
+oss.sonatype.org repository.
+
+Release the Staging Repository in Artifactory
+---------------------------------------------
+
+Release the artifact bundle in Artifactory:
+
+1. Login to https://oss.sonatype.org (you will need to use the same credentials you have
```
Comment (awholegunch) [inline RELEASING.rst]: Add trailing periods to these steps, so that they are all the same?

File: RELEASING.rst [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  git checkout develop
+  git merge master
+
+
+Announcing and Completing the Release
+=====================================
+
+Mark the release as complete in JIRA (in the Tephra Administration section):
+
+1. add a release for the next version if necessary
```
Comment (awholegunch) [inline RELEASING.rst]: change these steps to start with an initial capital letter, such as "Add a release..."

File: RELEASING.rst [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+including deploying the release artifacts to Maven repositories and the additional administrative
+steps to complete the release process.
+
+Prerequisites
+=============
+
+Maven Settings File
+-------------------
+
+Prior to performing a Tephra release, you must have an entry like the following in your
```
Comment (awholegunch) [inline RELEASING.rst]: you must have an entry such as this in your

File: RELEASING.rst (hunk 5 above)
Comment (awholegunch) [inline RELEASING.rst]: Add a release for the next version, if necessary

**Comment by awholegunch** [comment]
Minor comments, otherwise LGTM. I haven't performed this process, and can't vouch for its accuracy or completeness.

File: RELEASING.rst (hunk 2 above)
Comment (ghelmling) [inline RELEASING.rst]: Nope, the repo name appears as "comcontinuuity"

File: RELEASING.rst (hunk 4 above)
Comment (ghelmling) [inline RELEASING.rst]: Done

**Comment by ghelmling** [comment]
@awholegunch updated based on comments.

Luckily I have gone through this process a few times, based on a checklist that I turned into this doc.

**Comment by awholegunch** [comment]
LGTM.

**Pull request merged**